napari/components/viewer_model.py

```diff
@@ -558,12 +552,72 @@ def add_layer(self, layer: layers.Layer) -> layers.Layer:
         layer.name = self.layers._coerce_name(layer.name, layer)
         layer.events.name.connect(self.layers._update_name)
         layer.selected = True
-        self.layers.append(layer)
+        # For the labels layer we need to reset the undo/ redo
+        # history whenever the displayed slice changes. Once
+        # we have full undo/ redo functionality, this can be
+        # dropped.
+        if hasattr(layer, '_reset_history'):
+            self.dims.events.ndisplay.connect(layer._reset_history)
+            self.dims.events.order.connect(layer._reset_history)
+            self.dims.events.current_step.connect(layer._reset_history)
```

Comment:
why? Why reset history if no changes are made? I sometimes switch to 3d to verify some changes? 
I see that it is a copy of implementation, but It is not a good idea.

Reply:
Yeah, so this was to prevent copying from becoming too large, but has been confusing to users already (as discussed in #1448). This code will cause the current behavior not to change, but we might want to also address #1448 and improve the situation there too. Do you want to continue a discussion on #1448 how to make it better?

Reply:
@sofroniewn but have we ever observed this history becoming too large? I'm sort of inclined to remove this line, and *maybe* we can set a memory cap on the history and drop things when it exceeds it?

Reply:
it's also more complicated right now, because we don't store `_slice_indices` in the history, so when you move slice and you then did undo you'd paint the old slice incorrectly into the new slice. I think we should improve this, but dropping those lines will require more significant changes then I think belong in this PR